### PR TITLE
Refactor ViewModel LiveData

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Depends on #228

### CODE REVIEW

**Review by JorisBodin — CHANGES_REQUESTED** [review]

**Review by sirambd — CHANGES_REQUESTED** [review]
Waiting for changes

File: app/src/main/java/com/infomaniak/mail/ui/main/user/SwitchUserAccountsAdapter.kt [hunk 1]
```diff
@@ -100,14 +101,32 @@ class SwitchUserAccountsAdapter(
...
+    private fun createNewAccountsList(uiAccount: UiAccount): MutableList<UiAccount> {
+        val newList = accounts.toMutableList()
+        val existingAccount = newList.find { it.user.id == uiAccount.user.id }
+        if (existingAccount == null) {
+            newList.add(uiAccount)
+        } else {
+            existingAccount.mailboxes = uiAccount.mailboxes
+        }
+        return newList.sortAccounts()
+    }
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/main/user/SwitchUserAccountsAdapter.kt:122]: Refactor with a viewmodel

File: app/src/main/java/com/infomaniak/mail/ui/main/user/SwitchUserFragment.kt [hunk 2]
```diff
@@ -76,47 +73,27 @@ class SwitchUserFragment : Fragment() {
...
-                users.find { it.id == userId }?.let { user ->
-                    add(user to mailboxes)
+            with(switchUserViewModel) {
+                fetchUsersMailboxes(users)
+                users.forEach { user ->
+                    withContext(Dispatchers.Main) {
+                        mainViewModel.listenToMailboxes(user.id).observe(viewLifecycleOwner) { mailboxes ->
+                            onMailboxesChange(user, mailboxes)
+                        }
+                    }
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/ui/main/user/SwitchUserFragment.kt:89]: Refactor with a viewmodel

**Review by sirambd — CHANGES_REQUESTED** [review]
Waiting for request changes

File: app/src/main/java/com/infomaniak/mail/ui/main/user/SwitchUserAccountsAdapter.kt (hunk 1 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/user/SwitchUserAccountsAdapter.kt:122]: Done in #240

File: app/src/main/java/com/infomaniak/mail/ui/main/user/SwitchUserFragment.kt (hunk 2 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/user/SwitchUserFragment.kt:89]: Done in #240

**Pull request merged**